### ISSUE TITLE
Show free time in the agenda

### ISSUE THREAD

**Issue by gerazov** [nvim-orgmode/orgmode#96]
### Does this feature exist in Emacs orgmode core?

Yes

### Orgmode link

It seems to be built-in in e-macs.

![image](https://user-images.githubusercontent.com/15214418/134464095-cf12376b-6b4f-4c09-b9ca-e3c7b366cbfe.png)

### Feature value

It's really helpful when scheduling to-dos. And it doesn't need to crowd the agenda as in the printscreen above, i.e. a single dotted line between non-adjacent tasks would be really helpful :+1:

**Comment by kristijanhusak** [comment]
It's this option: https://www.gnu.org/software/emacs/manual/html_node/org/Time_002dof_002dday-specifications.html

I find it a bit annoying to be honest, but I guess someone can find it useful.

**Comment by gerazov** [comment]
Yeah it could be less obtrusive - a single line or smth, just to let you find a free time slot at a glimpse of the eye. Atm you have to check end times vs start times across the agenda ...

**Comment by kristijanhusak** [comment]
I'm not sure I understand how you mean to show only a single line?

**Comment by gerazov** [comment]
Smth like

![image](https://user-images.githubusercontent.com/15214418/134482244-136881a2-e27d-4fba-9e64-781e7e74adf0.png)

**Comment by kristijanhusak** [comment]
Is it configurable like this in the emacs itself or you just "draw" it for purposes of showing it to me? When I implement this (not sure when it will happen), it will follow the emacs orgmode functionality, which is he first screenshot I guess.

**Comment by gerazov** [comment]
I haven't used e-macs ...

**Comment by kristijanhusak** [comment]
Ah, ok. I just saw that dates in the screenshot are from 20219 😄 When it gets implemented, it will be as on the first screenshot, if that helps you at this moment at all.

**Comment by gerazov** [comment]
Sure - awesome :love_you_gesture:

**Comment by jgollenz** [comment]
@kristijanhusak 

I've started working on this. Currently, the `AgendaView` is built from `AgendaItem`s. Those separators required for the time grid 'are' `AgendaItem`s, but they definitely don't check all the boxes required for an `AgendaItem`. They don't have a real headline, but e.g. `build_agenda_item_content` expects the item to have one. Now, how do we deal with this? Especially since we may move to Teal. 

Do we want some kind of `SeparatorItem` type or should `AgendaItem` be modified to allow for more flexibility? Something like `AgendaItem:newSeparator`? Shall I 'spoof' an `AgendaItem` somehow?

**Comment by kristijanhusak** [comment]
> Something like AgendaItem:newSeparator

Go with this. I would treat these as agenda items, even though they are not a real headline.

**Comment by gerazov** [comment]
@jgollenz thanks for picking this one up :pray: 

It would be great if the free time is collapsible like in the second screenshot. This would avoid spamming the agenda if there are only a few scheduled items.

**Comment by jgollenz** [comment]
You're welcome :+1: 

> It would be great if the free time is collapsible

This sounds like a nice idea. However, it doesn't seem to be part of emacs orgmode. But it might be a case for a custom agenda view. For now, I'll focus on getting the default behavior right

**Comment by gerazov** [comment]
We could improve on emacs if it makes sense :thinking:

**Comment by jgollenz** [comment]
Agreed, but there are still a lot of things from the core we are missing. Those have higher prio for me ;)

**Comment by gerazov** [comment]
Ok, only if it's easy then :slightly_smiling_face: 

In my workflow I'm not really missing much functionality so I guessed we're nearing v1.0 :sunglasses:

**Comment by jgollenz** [comment]
Unfortunately we don't have a proper roadmap, so 'v1.0' isn't really defined :/

**Comment by cvigilv** [comment]
Any new insights into this? This is a feature of Orgmode that is pretty useful for scheduling and time-blocking. In Emacs, the option is `org-agenda-time-grid` ([source](https://orgmode.org/manual/Time_002dof_002dday-specifications.html)). I'm unsure if this could be implemented with the current API. If possible, let me know and I could come up with a small PR to implement this functionality.

**Comment by cvigilv** [comment]
Thanks for this! Already configured it and looks awesome. Will come back if I find anything out of place or bugs.

**Comment by MattCantor** [comment]
So awesome.  So far works great!

**Comment by gerazov** [comment]
Wow! That's perfect 🥳 Awesome job 😎
